src/main/java/servlets/ReceiptAnalysis.java

```diff
@@ -281,19 +283,64 @@ private static boolean isDate(String token) {
   }
 
   /**
-   * Checks the raw text in the builder for prices that can be parsed. The largest price found, if
-   * it exists, is added to the builder.
+   * Searches the raw text in the builder for the total price using multiple heuristics. If any find
+   * a price, it is added to the builder.
    */
   private static void checkForParsablePrices(AnalysisResults.Builder analysisBuilder) {
-    // Assume that the largest price on the receipt is the total price.
+    if (!findPriceAfterTotal(analysisBuilder)) {
+      findLargestPrice(analysisBuilder);
+    }
+  }
+
+  /**
+   * Checks the raw text in the builder for a price appearing after the word "total" and adds it to
+   * the builder if found. The return value indicates whether a price was added.
+   */
+  private static boolean findPriceAfterTotal(AnalysisResults.Builder analysisBuilder) {
+    ImmutableList<String> relevantTokens =
+        getTokensFromRawText(analysisBuilder.getRawText())
+            .filter(token -> isPrice(token) || containsTotal(token))
+            .collect(ImmutableList.toImmutableList());
+
+    double price = 0;
+    boolean priceFound = false;
+    boolean priceFoundAfterMostRecentTotal = true;
+
+    // Try to find the first price after each token containing "total", and keep the last one found.
```

Comment:
If you check the Macy's receipt I uploaded to the shared drive, this is an example of a receipt where it says "Total: 21.00" and further down it says "Your total savings is $6.25." Would this return $6.25? Maybe you can edit this heuristic or add another one to do a combination of highest price and searching for the word "total"

Reply:
As we discussed earlier, combining the two would fail in a case like "Your total before discounts are applied is $21.00, and your total after discounts is $17.75". I think it's best to stick with the current heuristics for the moment.